apps/prairielearn/src/lib/issues.ts

```diff
@@ -1,28 +1,30 @@
-// @ts-check
 import { eachSeries } from 'async';
 import { callbackify } from 'util';
 
 import * as sqldb from '@prairielearn/postgres';
 import { recursivelyTruncateStrings } from '@prairielearn/sanitize';
+import { Variant } from './db-types';
 
-/**
- * @typedef {Object} IssueData
- * @property {number | string} variantId
- * @property {string | null} studentMessage
- * @property {string | null} instructorMessage
- * @property {boolean} manuallyReported
- * @property {boolean} courseCaused
- * @property {Object} courseData
- * @property {Object} systemData
- * @property {number | string | null} authnUserId
- */
+interface IssueForErrorData {
+  variantId: string;
+  studentMessage: string | null;
+  courseData: Record<string, any>;
+  authnUserId: string | null;
```

Comment:
Nit: It looks like the only time an issue might not have an associated user (at least as far as TS will tell me) is when workspace file creation fails. Would it be worth it to just pass through a user ID to that point so that we can always require that this be set? (This is not a blocking comment.)

Reply:
That's an option, but the user id there is not that easy to retrieve, since the initialization code is called indirectly from a socket message... In any case, even if a user ID is presented there, I'd prefer to still keep the option of not providing a user ID in case future scenarios arise that require a non-user ID.

Reply:
Ah, good point. That's fair, we can leave this as nullable.